src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs

```diff
@@ -756,6 +763,12 @@ internal static string UnmangleMetadataNameForArity(string emittedTypeName, int
                         else
                         {
                             // This is a new child namespace
+                            if (cmp > 0)
+                            {
+                                // The sort order is violated for child namespace names. Obfuscation is the likely reason for this. 
```

Comment:
Where is this sort order documented?  It seems like we are relying here on the ordering of `StringComparer` to match up with that of metadata ordering.  Is that guaranteed to be the same?